xbmc/rendering/gl/RenderSystemGL.cpp

```diff
@@ -460,31 +460,13 @@ void CRenderSystemGL::Project(float &x, float &y, float &z)
 
 void CRenderSystemGL::CalculateMaxTexturesize()
 {
-  GLint width = 256;
-
-  // reset any previous GL errors
-  ResetGLErrors();
-
-  // max out at 2^(8+8)
-  for (int i = 0 ; i<8 ; i++)
+  GLint width;
+  glGetIntegerv(GL_MAX_TEXTURE_SIZE, &width);
+  m_maxTextureSize = width;
+  if (width > 65536) // have an upper limit in case driver acts stupid
```

Comment:
Should probably also check [`glGetError`](https://learn.microsoft.com/en-us/windows/win32/opengl/glgeterror)? Otherwise `width` could be uninitialized here.

Reply:
Maybe it would be better to initialize width to the default value of 2048 instead, since it seems unlikely this will fail. OpenGL docs mention that glGet* could return either GL_INVALID_ENUM (enum here is definitely valid) or GL_INVALID_VALUE which isn't applicable to this function

Reply:
Initializing to the default value is also fine with me. It's just the possibility of using an entirely initialized value that makes me uneasy :slightly_smiling_face: